cmd/lncli/cmd_invoice.go

```diff
@@ -2,13 +2,73 @@ package main
 
 import (
 	"encoding/hex"
+	"encoding/json"
 	"fmt"
 	"strconv"
 
 	"github.com/lightningnetwork/lnd/lnrpc"
 	"github.com/urfave/cli"
 )
 
+// newHopHint creates a new lnrpc.HopHint from the provided parameters.
+func newHopHint(nodeID string, chanID uint64, feeBaseMsat uint32,
+	feeProportionalMillionths uint32, cltvExpiryDelta uint32) (
+	*lnrpc.HopHint, error,
+) {
+
+	if nodeID == "" {
+		return nil, fmt.Errorf("node_id is missing in route hint")
+	}
+
+	if chanID == 0 {
+		return nil, fmt.Errorf("chan_id is missing or invalid in " +
+			"route hint")
+	}
+
+	return &lnrpc.HopHint{
+		NodeId:                    nodeID,
+		ChanId:                    chanID,
+		FeeBaseMsat:               feeBaseMsat,
+		FeeProportionalMillionths: feeProportionalMillionths,
+		CltvExpiryDelta:           cltvExpiryDelta,
+	}, nil
+}
+
+// parseRouteHints parses the route hints from a JSON string and returns a
+// slice of lnrpc.HopHint.
+func parseRouteHints(routeHints string) ([]*lnrpc.HopHint, error) {
+	var jsonHints []*lnrpc.HopHint
+	err := json.Unmarshal([]byte(routeHints), &jsonHints)
+
+	if err != nil {
+		return nil, fmt.Errorf("error parsing route hints JSON: "+
+			"%w", err)
+	}
+
+	if len(jsonHints) == 0 {
+		return nil, fmt.Errorf("no valid route hints found in JSON")
+	}
+
+	var invoiceHints []*lnrpc.HopHint
+	for _, jsonHint := range jsonHints {
+		newHint, err := newHopHint(
```

Comment:
we already have a `[]*lnrpc.HopHint` so why do we need to re-create a new set?

Reply:
if we defs need to do the validation here then can just:

```
func parseRouteHints(routeHints string) ([]*lnrpc.HopHint, error) {
	var hints []*lnrpc.HopHint
	if err := json.Unmarshal([]byte(routeHints), &hints); err != nil {
		return nil, fmt.Errorf("error parsing route hints JSON: %w",
			err)
	}

	if len(hints) == 0 {
		return nil, fmt.Errorf("no valid route hints found in JSON")
	}

	for _, hint := range hints {
		if hint.NodeId == "" {
			return nil, fmt.Errorf("node_id is missing in " +
				"route hint")
		}

		if hint.ChanId == 0 {
			return nil, fmt.Errorf("chan_id is missing or " +
				"invalid in route hint")
		}
	}

	return hints, nil
}
```

right?

Reply:
True! We could scrap the validation -- my original intent was to avoid hitting the RPC unless a few basic checks align, but can rely on the RPC for the validation 👍